phpunit/html/wp-html-walker-test.php

```diff
@@ -0,0 +1,469 @@
+	</div>
+</div>
+HTML;
+
+		$expected_output = <<<HTML
+<div data-details="{ &quot;key&quot;: &quot;value&quot; }" selected class="merge-message is-processed" checked>
+	<div class="select-menu d-inline-block">
+		<div checked class="MixedCaseHTML position-relative button-group Another-Mixed-Case" />
+		<div checked class="MixedCaseHTML position-relative button-group Another-Mixed-Case">
+			<button type="button" class="merge-box-button btn-group-merge rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Merge pull request
+			</button>
+
+			<button type="button" class="merge-box-button btn-group-squash rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Squash and merge
+			</button>
+
+			<button type="button"  aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Rebase and merge
+			</button>
+
+			<button aria-label="Select merge method" disabled="disabled" type="button" data-view-component="true" class="select-menu-button btn BtnGroup-item"></button>
+		</div>
+	</div>
+</div>
+HTML;
+
+		$w = new WP_HTML_Walker( $input );
+		$w->next_tag( 'div' );
+		$w->set_attribute( 'data-details', '{ "key": "value" }' );
+		$w->add_class( 'is-processed' );
+		$w->next_tag(
+			array(
+				'tag_name'   => 'div',
+				'class_name' => 'BtnGroup',
+			)
+		);
+		$w->remove_class( 'BtnGroup' );
+		$w->add_class( 'button-group' );
+		$w->add_class( 'Another-Mixed-Case' );
+		$w->next_tag(
+			array(
+				'tag_name'   => 'div',
+				'class_name' => 'BtnGroup',
+			)
+		);
+		$w->remove_class( 'BtnGroup' );
+		$w->add_class( 'button-group' );
+		$w->add_class( 'Another-Mixed-Case' );
+		$w->next_tag(
+			array(
+				'tag_name'     => 'button',
+				'class_name'   => 'btn',
+				'match_offset' => 2,
+			)
+		);
+		$w->remove_attribute( 'class' );
+		$w->next_tag( 'non-existent' );
+		$w->set_attribute( 'class', 'test' );
+		$this->assertSame( $expected_output, (string) $w );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_works_with_single_quote_marks() {
+		$w = new WP_HTML_Walker(
+			'<div id=\'first\'><span id=\'second\'>Text</span></div>'
+		);
+		$w->next_tag( array(
+			'tag_name' => 'div',
+			'id'       => 'first',
+		) );
+		$w->remove_attribute( 'id' );
+		$w->next_tag( array(
+			'tag_name' => 'span',
+			'id'       => 'second',
+		) );
+		$w->set_attribute( 'id', 'single-quote' );
+		$this->assertSame(
+			'<div ><span id="single-quote">Text</span></div>',
+			(string) $w
+		);
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_works_with_boolean_attributes() {
+		$w = new WP_HTML_Walker(
+			'<form action="/action_page.php"><input type="checkbox" name="vehicle" value="Bike"><label for="vehicle">I have a bike</label></form>'
+		);
+		$w->next_tag( 'input' );
+		$w->set_attribute( 'checked', true );
```

Comment:
This test fails as it produces `checked="1"` at the moment. The question is whether the test should get updated or the code should handle boolean values in some special way.

Reply:
Good point! I think this test should pass, these flag attributes signify `true` after all.